source/screens/caseScreens/CaseLogic.js

```diff
@@ -26,17 +26,30 @@ export const Status = {
 };
 const oldCaseLimit = 4 * 30 * 24 * 60 * 60 * 1000; // cases older than 4 months are classified as old.
 
+const getIds = async (formTypes, findFormByType) =>
+  formTypes.map(async type => {
+    const formSummary = await findFormByType(type);
+    return formSummary.id;
+  });
+
 /**
  * Takes a typeCase and an array of cases and returns [status, latestCase, relevantCases].
  * @param {obj} caseType an object with {name, forms: [formIds], icon: icon name, navigateTo: string with the navigation path}.
  * @param {[cases]} cases array of case objects.
  */
-export const getCaseTypeAndLatestCase = (caseType, cases) => {
+export const getCaseTypeAndLatestCase = async (caseType, cases, findFormByType) => {
   let latestUpdated = 0;
   let latestCase;
   const relevantCases = [];
+
+  const idPromises = await getIds(caseType.formTypes, findFormByType);
+  const formIds = [];
+  await Promise.all(idPromises).then(ids => {
+    ids.forEach(id => formIds.push(id));
+  });
```

Comment:
This snippet might be better to run in the getIds functions, so that function returns a list of ids to be used in here? 

So you could do this instead:
const ids = await getIds(caseType.formTypes, findFormByType); (
instead of 
const idPromises = await getIds(caseType.formTypes, findFormByType);